src/NHibernate/Id/GuidCombGenerator.cs

```diff
@@ -48,20 +50,19 @@ private Guid GenerateComb()
 
 			DateTime now = DateTime.UtcNow;
 
-			// Get the days and milliseconds which will be used to build the byte string 
-			TimeSpan days = new TimeSpan(now.Ticks - BaseDateTicks);
-			TimeSpan msecs = now.TimeOfDay;
+			// We use 2 bytes for the day (65,535 possible values) so we're good until 6/6/2079.
```

Comment:
Why would we be no more "good" after that? The days will have looped, and then what? There are 10 other bytes which remains mostly random. What trouble do you expect?